pkg/apis/core/types_shoot.go

```diff
@@ -154,6 +154,16 @@ type ShootStatus struct {
 	MigrationStartTime *metav1.Time
 	// Credentials contains information about the shoot credentials.
 	Credentials *ShootCredentials
+	// LastMaintenanceOperation holds information about the last maintenance operation on the Shoot.
+	LastMaintenanceOperation *LastMaintenanceOperation
```

Comment:
What do you think about using https://github.com/gardener/gardener/blob/017772a4526c5cf3b0f0469703f048f87e6fa211/pkg/apis/core/types_common.go#L105-L116 ?

This will allow to set the state of the operaiton, i.e. `Failed` or `Succeeded` which can be later shown in the Dashboard. 

Also, the maintenance operation status should considered when the shoot status label is computed here https://github.com/gardener/gardener/blob/cb5947fe332be9183500b089cffad2dd335e6a08/pkg/controllermanager/controller/shoot/statuslabel/reconciler.go#L52

Reply:
> What do you think about using

IIUC you want to enhance `LastOperation` so that it holds the maintenance information.

Reply:
@acumino I think @vpnachev's proposal was to extend the new `LastMaintenanceOperation` type with `LastOperation` so that it can contains information about whether the operation succeeded.

However, I think it's not quite accurate/a good fit since `LastOperation` contains too many unrelated fields.
What about extending `LastMaintenance` with 

```golang
 	// Status of the last operation, one of Aborted, Processing, Succeeded, Error, Failed. 
 	State LastOperationState 
```

only? Wouldn't this be good enough?

Reply:
Yes, IMO using LastOperation for LastMaintenanceOperation is not right, this will be too much duplication. Also in maintenance time we just update the shoot manifest and then shoot reconciliation takes over and does the rest of the task so having `Progress`, `LastOperationState`, and other fields in `LastMaintenanceOperation` doesn't make much sense to me.

> What about extending LastMaintenance with LastOperationState

For this one approach can be if update call for updating shoot in maintenance controller pass then it can be set to succeed and if not then set to false.

Reply:
> For this one approach can be if update call for updating shoot in maintenance controller pass then it can be set to succeed and if not then set to false.

Yes, that's the idea and one of the initial motivations for the backing issue, I think. We would like to know whether the last maintenances worked.

Reply:
There is no right way to figure out if the last maintenance was successful or not, cause we don't store any state of shoot in shoot status.

Reply:
OK, the maintenance controller only updates the spec, and we only know whether it "worked" when the `Shoot` was successfully reconciled after the `triggeredTime`. Fine, we can also get started without this `State` field for now, see how it goes and improve later on if needed.